features/storage/system_storage/SystemStorage.cpp

```diff
@@ -68,6 +83,32 @@ MBED_WEAK BlockDevice *BlockDevice::get_default_instance()
 
     return &default_bd;
 
+#elif COMPONENT_FLASHIAP
+
+    uint32_t top_address;
+    uint32_t bottom_address;
+    size_t area_size;
+    FlashIAP flash;
+    NVStore &nvstore = NVStore::get_instance();
+    nvstore.get_area_params(0, top_address, area_size); //Find where nvstore begins
```

Comment:
Does this work if NVStore is not present?

Reply:
No. However, if there is flash so there is also NVStroe.

Reply:
Is that true? I thought you need to call NVStore::init to create NVStore on disk.

Actually looking at [get_area_params](https://github.com/ARMmbed/mbed-os/blob/e7cd6ae2a98d7b24cf30072eef07b66740aa85bc/features/storage/nvstore/source/nvstore.cpp#L983), by calling this function we are indirectly formatting the disk with NVStore. Which means we also have to pull in the NVStore logic.

Reply:
With call to `nvstore.get_area_params()`:
| Module     |         .text |       .data |        .bss |
|------------|---------------|-------------|-------------|
| [fill]     |     131(+131) |       4(+4) |     17(+17) |
| [lib]      | 20671(+20671) | 2488(+2488) |   117(+117) |
| components |     902(+902) |       0(+0) |       0(+0) |
| drivers    |     764(+764) |       4(+4) |   140(+140) |
| features   |   2530(+2530) |       0(+0) |   128(+128) |
| hal        |   1460(+1460) |       4(+4) |     68(+68) |
| main.o     |       10(+10) |       0(+0) |       0(+0) |
| platform   |   1624(+1624) |   256(+256) |   133(+133) |
| rtos       |   8293(+8293) |   168(+168) | 6053(+6053) |
| targets    | 11687(+11687) |       8(+8) |   912(+912) |
| Subtotals  | 48072(+48072) | 2932(+2932) | 7568(+7568) |

Without call to `nvstore.get_area_params()`:
| Module     |        .text |    .data |      .bss |
|------------|--------------|----------|-----------|
| [fill]     |      123(-8) |    4(+0) |    21(+4) |
| [lib]      |    20671(+0) | 2488(+0) |   117(+0) |
| components |      902(+0) |    0(+0) |     0(+0) |
| drivers    |      764(+0) |    4(+0) |   140(+0) |
| features   |   140(-2390) |    0(+0) |   44(-84) |
| hal        |     1460(+0) |    4(+0) |    68(+0) |
| main.o     |       10(+0) |    0(+0) |     0(+0) |
| platform   |     1620(-4) |  256(+0) |   133(+0) |
| rtos       |   8055(-238) |  168(+0) |  6053(+0) |
| targets    |    11687(+0) |    8(+0) |   912(+0) |
| Subtotals  | 45432(-2640) | 2932(+0) | 7488(-80) |

That's an extra 2.6KiB on platforms that just want internal flash not NVStore. Could we instead use config options specify the flash region? That's what we're currently doing for SPIF and SD pins.

Reply:
I have removed nvstore and added back the configuration option. 
If no configuration supplied and still FlashIAP block device is enabled the block device will take the address of the first sector after code section till the end of the flash.

Reply:
Thanks for that 👍